### PR TITLE
feat(sudo): add service for retrieving generic data from our DB

### LINKED ISSUE #57: Service for retrieving generic data from our DB
**Description**

- `Get` and `List` for a specified service (string/enum, e.g. "course") and optionally a type (string, e.g. "courseSeries") returning JSON
- Access to metadata (events, permissions, etc.)

### PR DESCRIPTION
**Pull request by JonasWanke**
**Fixes: #57**

**Checklist**

- [x] Commit messages follow conventionalcommits.org

**Changes**

### CODE REVIEW

File: hpi/cloud/sudo/v1test/sudo_service.proto [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+  int32 page_size = 1;
+
+  // A pagination token returned from a previous call to `ListEntities`
+  // that indicates where this listing should continue from.
+  string page_token = 2;
+
+  // Required. The service (e.g. "news") whose entities should be returned.
+  string service = 3;
+
+  // If set, only entities of this type will be returned.
```
Comment (ShirleyNekoDev) [inline hpi/cloud/sudo/v1test/sudo_service.proto]: There is no reason to get all entities with different types.

File: hpi/cloud/sudo/v1test/sudo_service.proto [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+  rpc UpdatePermissions (UpdatePermissionsRequest) returns (Entity);
+
+  // Updates the deleted status of an entity.
+  rpc UpdateDeleted (UpdateDeletedRequest) returns (Entity);
+
+  // Updates the published status of an entity.
+  rpc UpdatePublished (UpdatePublishedRequest) returns (Entity);
+}
+
+
```
Comment (ShirleyNekoDev) [inline hpi/cloud/sudo/v1test/sudo_service.proto:49]: ListEntityTypes missing

File: hpi/cloud/sudo/v1test/sudo_service.proto [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+  // Required. The entity ID.
+  string id = 2;
+
+  // Required. The new value.
+  string new_value = 3;
+}
+
+
+// Request message for adding a keep-alive event to the entity using [UpdateKeepAlive][hpi.cloud.sudo.v1test.SudoService.UpdateKeepAlive].
+message UpdateKeepAliveRequest {
```
Comment (ShirleyNekoDev) [inline hpi/cloud/sudo/v1test/sudo_service.proto]: Idk if needed

**Review by ShirleyNekoDev — APPROVED** [review]

File: hpi/cloud/sudo/v1test/sudo_service.proto (hunk 3 above)
Comment (JonasWanke) [reply hpi/cloud/sudo/v1test/sudo_service.proto]: I suggest we talk about KeepAlives (whether we need them or if there's a better alternative) on the next meeting

Comment (JonasWanke) [reply hpi/cloud/sudo/v1test/sudo_service.proto]: We decided to remove this event type. Instead, we handle this as a special case of an UpdateValueEvent: If nothing changed, the payload stays empty.

**Comment by JonasWanke** [comment]
And a small fix: Some request methods were missing the type parameter. As entities are identified by a (service, type, id)-tuple, those were added.

**Comment by JonasWanke** [comment]
@ShirleyNekoDev Can we merge this?

**Review by ShirleyNekoDev — APPROVED** [review]

**Pull request merged**